utils/predicate/predicate_bytes.go

```diff
@@ -0,0 +1,68 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package predicateutils
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// PredicateEndByte is used as a delimiter for the bytes packed into a precompile predicate.
+// Precompile predicates are encoded in the Access List of transactions in the access tuples
+// which means that its length must be a multiple of 32 (common.HashLength).
+// For messages with a length that does not comply to that, this delimiter is used to
+// append/remove padding.
+var PredicateEndByte = byte(0xff)
+
+// PackPredicate packs [predicate] by delimiting the actual message with [PredicateEndByte]
+// and zero padding to reach a length that is a multiple of 32.
+func PackPredicate(predicate []byte) []byte {
+	predicate = append(predicate, PredicateEndByte)
+	return common.RightPadBytes(predicate, (len(predicate)+31)/32*32)
```

Comment:
can we use `common.HashLength` instead of using hardcoded `32`?

Reply:
also in the rest of the file

Reply:
This is copying the style used here: https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/pack.go#L33. I think it's clearer to use 32 in this case since we're using 31 as well, so we can either have the current code or:

always use constant and relative constant
```
return common.RightPadBytes(predicate, (len(predicate)+(common.HashLength-1))/common.HashLength*common.HashLength)
```

use 31 but the constant for 32
```
return common.RightPadBytes(predicate, (len(predicate)+31)/common.HashLength*common.HashLength)
```

Reply:
if these are supposed to be same with hash length then I'd say we should (and geth should) use constant. but no pressure.